### PR TITLE
Replacing deprecated constants and adding story in SnapInstallWarning

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- This PR is a part of #19526 
- #18714 

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/28399fb2-c074-4067-a233-cb719cd40397

https://github.com/MetaMask/metamask-extension/assets/8112138/78aae6be-794a-4f3e-b00b-8a672eb2b5b4

Screenshot 2023-08-23 at 9 16 16 PM

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by dhruvv173** [comment]
the refactoring was done by @georgewrmarshall [`here`](https://github.com/MetaMask/metamask-extension/pull/19584/commits/ec0e9ff5a29dc1a801edb72f3be53b90a9863755)

File: ui/components/app/snaps/snap-install-warning/snap-install-warning.js [hunk 1]
```diff
@@ -29,118 +41,117 @@ const checkboxStateReducer = produce((state, action) => {
...
+              display={Display.Flex}
+              gap={2}
+              alignItems={AlignItems.center}
+            >
+              <CheckBox
+                checked={checkboxState[warning.id] ?? false}
+                id={warning.id}
+                onClick={() => onCheckboxClicked(warning.id)}
+              />
+              <Label htmlFor={warning.id}>{warning.message}</Label>
```
Comment (georgewrmarshall) [inline ui/components/app/snaps/snap-install-warning/snap-install-warning.js:134]: we recently merged a new checkbox component so you could update this as well to use the one from the component-library. The component props are different so make sure to read the docs https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-checkbox--docs

Comment (dhruvv173) [reply ui/components/app/snaps/snap-install-warning/snap-install-warning.js]: updated screenshots and made changes [`here`](https://github.com/MetaMask/metamask-extension/pull/20284/commits/a16289568d1dd752815167fadfb16bb5a0abdedd)

**Review by Ameralameri — APPROVED** [review]
Approv

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, looks like there are some failing e2e tests could you please resolve and I can review

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Thanks for your contribution @dhruvv173

**Review by georgewrmarshall — DISMISSED** [review]

**Review by georgewrmarshall — DISMISSED** [review]

**Review by garrettbear — DISMISSED** [review]

**Review by eriknson — DISMISSED** [review]

File: ui/components/app/snaps/snap-install-warning/snap-install-warning.js [hunk 2]
```diff
@@ -29,118 +41,117 @@ const checkboxStateReducer = produce((state, action) => {
...
+          display={Display.Flex}
+          flexDirection={FlexDirection.Column}
+          gap={4}
+          marginBottom={4}
+        >
+          {warnings.map((warning) => (
+            <Box
+              key={warning.id}
+              display={Display.Flex}
+              gap={2}
```
Comment (eriknson) [inline ui/components/app/snaps/snap-install-warning/snap-install-warning.js:126]: If `gap` represents the padding / margin between the checkbox and the label, we should try increasing it to **4** or potentially **6**

cc @david0xd

File: ui/components/app/snaps/snap-install-warning/snap-install-warning.js [hunk 3]
```diff
@@ -29,118 +41,117 @@ const checkboxStateReducer = produce((state, action) => {
...
+          <Text variant={TextVariant.headingMd} textAlign={TextAlign.Center}>
+            {t('snapInstallWarningHeading')}
+          </Text>
+        </ModalHeader>
+        <Text textAlign={TextAlign.Center} marginBottom={4}>
+          {warnings.length > 1
+            ? t('snapInstallWarningCheckPlural', [
+                <Text
+                  key="snapNameInWarningDescription"
+                  fontWeight={FontWeight.Medium}
```
Comment (eriknson) [inline ui/components/app/snaps/snap-install-warning/snap-install-warning.js:100]: `fontWeight` should be **regular** for the label except for the snap's name which should be **medium**

**Review by eriknson — COMMENTED** [review]
I suggest @david0xd to do a final review before merging this

File: ui/components/app/snaps/snap-install-warning/snap-install-warning.js (hunk 2 above)
Comment (david0xd) [reply ui/components/app/snaps/snap-install-warning/snap-install-warning.js:126]: I think it should match the previous version (at the best possible level).

This is on the latest Flask published:
Screenshot 2023-08-25 at 10 29 03

And on this PR:
Screenshot 2023-08-25 at 10 27 25

Comment (georgewrmarshall) [reply ui/components/app/snaps/snap-install-warning/snap-install-warning.js:125]: Good catch with the checkbox and label spacing and alignment. It should be aligned to the top and the spacing should be 16px(`4`) I've updated it.

File: ui/components/app/snaps/snap-install-warning/snap-install-warning.js [hunk 4]
```diff
@@ -29,118 +40,117 @@ const checkboxStateReducer = produce((state, action) => {
...
-            ])
-          : t('snapInstallWarningCheck', [
-              <Text
-                key="snapNameInWarningDescription"
-                fontWeight={FontWeight.Medium}
-                as="span"
-              >
-                {snapName}
-              </Text>,
-            ])}
```
Comment (georgewrmarshall) [inline ui/components/app/snaps/snap-install-warning/snap-install-warning.js:120]: This code hasn't been changed cc @eriknson @david0xd

File: ui/components/app/snaps/snap-install-warning/snap-install-warning.js (hunk 3 above)
Comment (georgewrmarshall) [reply ui/components/app/snaps/snap-install-warning/snap-install-warning.js:99]: Right on! I've updated this as well

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, would it be possible to resolve the conflicts in this PR?

**Pull request closed without merging**